hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java

```diff
@@ -88,19 +93,25 @@ private boolean isContainerReplicatingOrDeleting(ContainerID containerID) {
    * 5. Container should be closed.
    * 6. If the {@link LegacyReplicationManager} is enabled, then the container should not be an EC container.
    * @param node DatanodeDetails for which to find candidate containers.
-   * @return NavigableSet of candidate containers that satisfy the criteria.
+   * @return Set of candidate containers that satisfy the criteria.
    */
-  public NavigableSet<ContainerID> getCandidateContainers(
+  public Set<ContainerID> getCandidateContainers(
       DatanodeDetails node, long sizeMovedAlready) {
-    NavigableSet<ContainerID> containerIDSet =
-        new TreeSet<>(orderContainersByUsedBytes().reversed());
     try {
-      containerIDSet.addAll(nodeManager.getContainers(node));
+      // Initialize containerSet for node
+      if (!setMap.containsKey(node)) {
```

Comment:
I am trying to understand why do we need order based on containerUsed Size? Do largest size container moving is efficient? or it can be random container which can be moved?
we do not have statistics and always choose greatest container size only.  IMO, random can be better candidate in this for movement.